libyul/AsmParser.cpp

```diff
@@ -494,24 +486,9 @@ TypedName Parser::parseTypedName()
 YulString Parser::expectAsmIdentifier()
 {
 	YulString name{currentLiteral()};
-	switch (currentToken())
-	{
-	case Token::Return:
-	case Token::Byte:
-	case Token::Address:
-	case Token::Bool:
-	case Token::Identifier:
-	case Token::Var:
-	case Token::In:
-		break;
-	default:
-		expectToken(Token::Identifier);
-		break;
-	}
-
 	if (m_dialect.builtin(name))
```

Comment:
Should this be changed to `if (currentToken() == Token::Identifier && m_dialect.builtin(name))` ?

Since we have `expectToken` afterwards to keep the correct sourcelocation?

Reply:
yea that sounds safer

Reply:
Changed.